libsolidity/analysis/TypeChecker.cpp

```diff
@@ -113,18 +113,24 @@ void TypeChecker::checkDoubleStorageAssignment(Assignment const& _assignment)
 	size_t storageByteArrayPushes = 0;
 	size_t storageByteAccesses = 0;
 	auto count = [&](TupleExpression const& _lhs, TupleType const& _rhs, auto _recurse) -> void {
-		TupleType const& lhsType = dynamic_cast<TupleType const&>(*type(_lhs));
-		TupleExpression const* lhsResolved = dynamic_cast<TupleExpression const*>(resolveOuterUnaryTuples(&_lhs));
+		auto& lhsType = dynamic_cast<TupleType const&>(*type(_lhs));
+		auto* lhsResolved = dynamic_cast<TupleExpression const*>(resolveOuterUnaryTuples(&_lhs));
```

Comment:
Our convention is to keep the explicit `const` when using `auto` for clarity:
```suggestion
		auto const& lhsType = dynamic_cast<TupleType const&>(*type(_lhs));
		auto const* lhsResolved = dynamic_cast<TupleExpression const*>(resolveOuterUnaryTuples(&_lhs));
```